docs/Solana/DataFeeds/using-data-feeds-off-chain.md

```diff
@@ -62,34 +80,150 @@ After you install the required tools, clone the example code from the [solana-st
 
     The example code retrieves and prints the current price feed data until you close the application:
 
     ```
-    yarn run v1.22.18
-    $ ts-node ./read-data.ts
-    9731000000
-    9732000000
-    9730839909
-    9734000000
+    4027000000
+    4026439929
+    4026476542
+    4023000000
+    ```
+
+To learn more about Solana and Anchor, see the [Solana Documentation](https://docs.solana.com/) and the [Anchor Documentation](https://project-serum.github.io/anchor/).
+
+## Adding Data Feeds to an existing off-chain project
+
+You can read Chainlink Data Feed data off-chain in your existing project by using the [Chainlink Solana NPM library](https://www.npmjs.com/package/@chainlink/solana-sdk).
+
+> 🚧 Reading feed data
+>
+> Although you can directly query the data feed accounts, you should not rely on the memory layout always being the same as it currently is. Based on this, the recommendation is to always use the consumer library.
+
+### Install the required tools
+
+Before you begin, set up your environment. Any of these steps can be skipped if you have already done them as part of your project:
+
+1. Install [Node.js 14 or higher](https://nodejs.org/en/download/). Run `node --version` to verify which version you have installed:
+
+    ```sh
+    node --version
+    ```
+
+1. Change to your project directory or create a new directory.
+
+    ```sh
+    mkdir off-chain-project && cd off-chain-project
+    ```
+
+1. Optionally [install Yarn](https://classic.yarnpkg.com/lang/en/docs/install/) to use as a package manager and initialize yarn if your project does not already have a `package.json` file:
+
+    ```sh
+    npm install -g yarn && yarn init
+    ```
+
+1. Add the [Anchor library](https://www.npmjs.com/package/@project-serum/anchor) to your project:
+
+    ```sh yarn
+    yarn add @project-serum/anchor
+    ```
+    ```sh npm
+    npm i @project-serum/anchor
+    ```
+
+1. Add the [Chainlink Solana NPM library](https://www.npmjs.com/package/@chainlink/solana-sdk) to your project:
+
+    ```sh yarn
+    yarn add @chainlink/solana-sdk
+    ```
+    ```sh npm
+    npm i -g @chainlink/solana-sdk
+    ```
+
+1. Create a temporary Solana wallet to use for this example. Alternatively, if you have an existing wallet that you want to use, locate the path to your [keypair](https://docs.solana.com/terminology#keypair) file and use it as the keypair for the rest of this guide.
+
+    ```sh
+    solana-keygen new --outfile ./id.json
+    ```
+
+1. Set the [Anchor environment variables](https://www.twilio.com/blog/2017/01/how-to-set-environment-variables.html). Anchor uses these to determine which wallet to use and how to get a connection to a Solana cluster. Because this example does not generate or sign any transactions, no SOL tokens are required. The wallet is required only by the Anchor library. For a list of available networks and endpoints, see the [Solana Cluster RPC Endpoints](https://docs.solana.com/cluster/rpc-endpoints) documentation.
```

Comment:
```suggestion
1. Set the [Anchor environment variables](https://www.twilio.com/blog/2017/01/how-to-set-environment-variables.html). Anchor uses these to determine which wallet to use and how to get a connection to a Solana cluster. Because this example does not generate or sign any transactions, no lamports are required. The wallet is required only by the Anchor library. For a list of available networks and endpoints, see the [Solana Cluster RPC Endpoints](https://docs.solana.com/cluster/rpc-endpoints) documentation.
```